src/core-sdk-tasks/DownloadFile.cs

```diff
@@ -39,31 +54,70 @@ public override bool Execute()
                 var filePath = Uri.Substring(FileUriProtocol.Length);
                 Log.LogMessage($"Copying '{filePath}' to '{DestinationPath}'");
                 File.Copy(filePath, DestinationPath);
+                return true;
             }
-            else
+
+            var lastStatus = await DownloadWithRetries(Uri, DestinationPath);
```

Comment:
Use explicit `bool` since the type isn't obvious on the right side. I also think `success` would be a simpler name:

```suggestion
            bool success = await DownloadWithRetries(Uri, DestinationPath);
```

Reply:
Oh wait. Use explicit `bool?`. This was really confusing!